code/controllers/subsystems/falling.dm

```diff
@@ -0,0 +1,103 @@
+#define REMOVE_AND_CONTINUE			\
+	falling -= victim;				\
+	victim.multiz_falling = 0;		\
+	if (MC_TICK_CHECK) return;		\
+	continue;
+
+/var/datum/controller/subsystem/falling/SSfalling
+
+/datum/controller/subsystem/falling
+	name = "Falling"
+	flags = SS_NO_INIT | SS_TICKER | SS_KEEP_TIMING
+	wait = 1
+
+	var/list/falling = list()
+	var/list/currentrun
+
+/datum/controller/subsystem/falling/New()
+	NEW_SS_GLOBAL(SSfalling)
+
+/datum/controller/subsystem/falling/stat_entry()
+	..("F:[falling.len]")
+
+/datum/controller/subsystem/falling/fire(resumed = 0)
+	if (!resumed)
+		currentrun = falling.Copy()
+
+	var/list/curr = currentrun
+
+	while (curr.len)
+		var/atom/movable/victim = curr[curr.len--]
+
+		if (QDELETED(victim))
+			REMOVE_AND_CONTINUE
```

Comment:
Oh, it will. That's what I get for fucking with defines after implementing them.